### ISSUE TITLE
Pure Cool Link not registering as sensors

### ISSUE THREAD

**Issue by kingofzeal** [shenxn/ha-dyson#134]
Updated to 0.16.4-4, but I started seeing the following error in the logs:

```
2022-04-29 09:00:30 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform dyson_local.sensor: Platform not found (cannot import name 'SensorDeviceClass' from 'homeassistant.components.sensor' (/usr/src/homeassistant/homeassistant/components/sensor/__init__.py)).
2022-04-29 09:00:30 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform dyson_local.switch: Platform not found (cannot import name 'EntityCategory' from 'homeassistant.helpers.entity' (/usr/src/homeassistant/homeassistant/helpers/entity.py)).
```

And I can no longer set the fan speed, nor is the temperature reporting - I can only turn the fan on or off.

I've attempted to roll back to all available previous versions, but none of them seem to work. I have tried to reinstall the device within Home Assistant and have also gone so far as to remove the integration and HACS repository and start from scratch, but I still have the same issues.

I am not sure how to fix this issue. Is there any guidance for this type of issue?

**Comment by Kakise** [comment]
Hello,
Can you give me more logs ? These 2 lines don't really help me pinpoint the issue. Also, what's your device ?

**Comment by wormoworm** [comment]
In case it helps, I was having a similar issue (the same two log lines as @kingofzeal posted, although I was able to set the fan speed).

I was able to solve this by updating Home Assistant to the latest version (2022.5.3).

**Comment by kingofzeal** [comment]
I apologize for not getting back to this sooner.

Unfortunately, I am not able to provide additional logging - the lines provided are the only information about the issue I was able to find in any of the logs. I'm running Home Assistant in Docker (specifically the `homeassistant/raspberrypi3-homeassistant` image), and the issue also resolved for me when I updated the container to the most recent version (I do not know explicitly which version of HA I was running previously - likely 2021.x, but I am unable to find the image in Docker hub anymore so I can't provide an exact version tag). I also don't know which version of ha-dyson I was running prior to the update but was probably 0.14 or 0.15 (neither of which show as options in HACS, so I was unable to roll back to test if they worked).

If there's not enough information here to go on, feel free to close. At the very least if others see and search for the same messages, they can learn that an update to HA appears to fix it.
